app/src/main/java/net/squanchy/navigation/RoutingActivity.java

```diff
@@ -0,0 +1,31 @@
+package net.squanchy.navigation;
+
+import android.content.Intent;
+
+import net.squanchy.fonts.TypefaceStyleableActivity;
+import net.squanchy.navigation.deeplink.DeepLinkRouter;
+
+import timber.log.Timber;
+
+public class RoutingActivity extends TypefaceStyleableActivity {
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        RoutingComponent component = RoutingInjector.obtain(this);
+        DeepLinkRouter deepLinkRouter = component.deepLinkRouter();
+        Navigator navigator = component.navigator();
+
+        Intent intent = getIntent();
+        if (deepLinkRouter.hasDeepLink(intent)) {
```

Comment:
Can we have `deepLinkRouter` return an optional of the deeplink, so we don't leak details for where the deeplink is inside the intent on this `if` branch?